src/http.ts

```diff
@@ -30,6 +39,131 @@ export interface SourceImageOptions {
   requestEtag?: string
 }
 
+interface UsageTrackingItem {
+  runningCount: number;
+  cacheKey: string;
+  lastAccess: number;
+  inputCacheFile: string
+  size: number
+}
+
+type UsageTracking = Record<
+  string,
+  UsageTrackingItem
+>;
+
+const USAGE_TRACKING_KEY = 'usage-tracking'
+const trackingLock = new Lock()
+
+async function getTracking (metadataStore: Storage): Promise<UsageTracking> {
+  return ((await metadataStore.getItem(USAGE_TRACKING_KEY)) as
+      | UsageTracking
+      | undefined) ?? {}
```

Comment:
Since `Promise<UsageTracking>` is being returned can't the `| undefined` can be omitted?

```suggestion
  return ((await metadataStore.getItem(USAGE_TRACKING_KEY)) ??
    {}) as UsageTracking
```